backend/service/chaos/experimentation/experimentstore/experiment_specification.go

```diff
@@ -0,0 +1,91 @@
+package experimentstore
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	experimentationv1 "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	"github.com/lyft/clutch/backend/id"
+)
+
+const runIdRegExp = `^[A-Za-z0-9-._~]+$`
+
+type ExperimentSpecification struct {
+	RunId     string
+	ConfigId  string
+	StartTime time.Time
+	EndTime   *time.Time
+	Config    *any.Any
+}
+
+// New returns a new experimentationSpecification instance.
+func NewExperimentSpecification(ced *experimentationv1.CreateExperimentData, now time.Time) (*ExperimentSpecification, error) {
+	var runId = ced.RunId
+	// If the run identifier is not provided, generate one
+	if runId == "" {
+		runId = id.NewID().String()
+	} else if !regexp.MustCompile(runIdRegExp).MatchString(runId) {
+		return nil, status.Errorf(codes.InvalidArgument, "provided experiment runId (%v) contained unallowed characters and was not matched by \"%v\" regular expresion", runId, runIdRegExp)
+	}
+
+	if ced.Config == nil {
+		return nil, status.Error(codes.InvalidArgument, "experiment config cannot be equal to nil")
+	}
```

Comment:
do we support pgv validations? Would be nice to be able to just specify this on the IDL

Reply:
Done 👌